dask/array/gufunc.py

```diff
@@ -144,9 +147,18 @@ def apply_gufunc(func, signature, *args, **kwargs):
 
     ## Assert output_dtypes
     if output_dtypes is None:
-        raise ValueError("Must specify `output_dtypes` of output array(s)")
-    elif isinstance(output_dtypes, str):
-        otypes = list(output_dtypes)
+        output_dtypes = apply_infer_dtype(func, args, kwargs, "apply_gufunc", "output_dtypes", nout)
+
+    if isinstance(output_dtypes, str):
```

Comment:
I don't think it's worth the complexity of supporting strings as lists of typecodes. I would stick with just promoting `dtype='float'` to `dtype=['float']`. I'll grant that writing `dtype='BOOL'` seems unlikely but there's no guarantee conflicts will not arise in the future.

Reply:
Good point - I've kept this behavior from our [original discussion](https://github.com/dask/dask/pull/3109#discussion_r188333528) in PR #3109 where it was pointed out that `np.vectorize` allows the abbreviated syntax `np.vectorize(..., otypes='ff')`

From my side we could get rid of this option. Nevertheless on a broader perspective: `signature` argument is a string (with comma delimiters) to keep notation short and handy. Allowing `output_dtypes` only to be lists (in case of multiple outputs) is a bigger syntax than a handy string notation, which includes all dtypes. Therefore I could imaging at least providing a short "one string" notation, by using comma delimiter, just as in `signature`, i.e. `apply_gufunc(..., signature="()->(),()", output_dtypes="f,f")`. Any thoughts?

Reply:
...of course a list of dtypes would be possible as well, besides the suggested one string notation.

Reply:
I think we should make a choice between supporting syntax like `output_dtypes='ff'` and `output_dtypes='float'`/`output_dtypes=np.float64`. Either strings should be consistently considered either iterables of characters (like np.vectorize) or the string name of a single dtype. I'm OK either way.

Reply:
...just for comparison I've pushed the comma-delimted version. Am understanding it would introduce yet another syntax. I am going to get back at this issue tomorrow...